lib/kast.js

```diff
@@ -67,6 +67,56 @@ const joinMatches = (match1, match2) => {
 const antimatch = (pattern, term, path = []) => {
   if (pattern.node == KVARIABLE) {
     return [];
+  } else if (pattern.node == KAPPLY && term.node == KAPPLY && pattern.label == term.label && pattern.label == "_Map_") {
+    // gather keys lhs
+    const lhs = pattern.args
+      .filter(pair => pair.node !== "KVariable")
+      .map(pair => [format(pair.args[0]), pair.args[1]])
+    const rhs = term.args
+      .filter(pair => pair.node !== "KVariable")
+      .map(pair => [format(pair.args[0]), pair.args[1]])
+    const lhs_keys = lhs.map(pair => pair[0]);
+    const rhs_keys = rhs.map(pair => pair[0]);
+    // TODO - introduce a type arg which states if arg is missing or rejected
```

Comment:
What's the consequence of not having this implemented now? Don't quite understand what a "type arg" is.

In general I think we should strive not to have TODOs in the code that are not mirrored somewhere (for visibility, and to have more room to explain what the TODO is about). But of course one can wait until this PR is merged to create an issue :)

Reply:
Damn, I need to clean my code more
This was a comment for me as I stopped for lunch here to not forget what should be coded on next, everything was implemented after lunch
lunch was also delicious